### PR TITLE
Sindhu/model accuracy verification 

### PR DESCRIPTION
**Pull request by sindhu-nervana**

### CODE REVIEW

File: diagnostics/model_accuracy/verify_inference_model.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    for i, d in enumerate(data):
+        if (model in data[i]["model_name"]):
+            # Tolerance check
+            diff = abs(float(top1_accuracy) - float(data[i]["accuracy"]))
+            if (diff <= 0.001):
+                print("\nResult:Model Accuracy is expected for " + model + ' ' +
+                      top1_accuracy)
+            else:
+                print("\nResult:Model Accuracy is not as expected for " +
```
Comment (avijit-nervana) [inline diagnostics/model_accuracy/verify_inference_model.py]: The check should fail i.e., throw an exception with details of the error.

File: diagnostics/model_accuracy/verify_inference_model.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                      model + ' ' + top1_accuracy)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Accuracy verification for TF models using ngraph.')
+
+    parser.add_argument('--model_name', help='Model name to run inference')
+    cwd = os.getcwd()
+    # This script must be run from this location
```
Comment (avijit-nervana) [inline diagnostics/model_accuracy/verify_inference_model.py]: Why? Please remove this restriction.
You can add an optional parameter specifying the source of the model repository location on disk. If not then the local directory will be used.

File: diagnostics/model_accuracy/verify_inference_model.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            #TODO:SSD-MobileNet-v1 is not work, will re-enable this after debugging
+            # if(data[i]["model_type"] == "Object Detection"):
+            #     CHECKPOINT = data[i]["CHECKPOINT"]
+            #     command_executor('git apply ' + pwd + '/object_detection.patch')
+            #     os.chdir(pwd + "/models/research/")
+            #     command_executor("python setup.py install")
+            #     command_executor("python slim/setup.py install")
+            #     command_executor("export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`/object_detection:`pwd`/slim")
+    try:
+        if (model_name in "Inception_v4"):
```
Comment (avijit-nervana) [inline diagnostics/model_accuracy/verify_inference_model.py]: Read all of the test parameters from the JSON data structure.

**Comment by sindhu-nervana** [comment]
@avijit-nervana I made changes as per your comments, please review. Thanks.

File: diagnostics/model_accuracy/verify_inference_model.py [hunk 4]
```diff
@@ -39,69 +39,71 @@ def download_repo(repo, target_name=None, version='master'):
...
+                if models_dir is None:
+                    os.chdir(pwd + "/models/research/slim")
+                else:
+                    os.chdir(models_dir + "research/slim")
                 command_executor("export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`")
                 command_executor('git apply ' + pwd +
                                  '/image_recognition.patch')
 
-            #TODO:SSD-MobileNet-v1 is not work, will re-enable this after debugging
+            #TODO:SSD-MobileNet-v1 does not run, will re-enable this after debugging
```
Comment (avijit-nervana) [inline diagnostics/model_accuracy/verify_inference_model.py]: Remove dead code.

File: diagnostics/model_accuracy/verify_inference_model.py [hunk 5]
```diff
@@ -39,69 +39,71 @@ def download_repo(repo, target_name=None, version='master'):
...
+def run_inference(model_name, models_dir):
+    parameters = \
+    '[{"model_type" : "Image Recognition", "model_name" : "Inception_v4", \
+    "cmd" : "OMP_NUM_THREADS=28 KMP_AFFINITY=granularity=fine,compact,1,0 python eval_image_classifier.py \
+    --alsologtostderr --checkpoint_path=/nfs/site/home/skantama/validation/models/research/checkpoints/inception_v4.ckpt \
```
Comment (avijit-nervana) [inline diagnostics/model_accuracy/verify_inference_model.py]: Indent the command fields for better readability.

Comment (sindhu-nervana) [reply diagnostics/model_accuracy/verify_inference_model.py]: made these changes.

**Pull request merged**